src/common/config.rs

```diff
@@ -0,0 +1,366 @@
+              fn try_from(value: &Kind) -> Result<Self, Self::Error> {
+                  match value {
+                      Kind::NullValue(v) => Ok(*v as $t),
+                      Kind::NumberValue(v) => Ok(*v as $t),
+                      Kind::StringValueStatic(v) => Ok(v.parse::<$t>()?),
+                      _ => Err(AttributeError::ConversionImpossibleError),
+                  }
+              }
+          }
+        )*
+    }
+}
+
+floats!(f64, f32);
+
+impl<V> TryFrom<Kind> for BTreeMap<&'static str, V>
+where
+    V: for<'a> std::convert::TryFrom<&'a Kind, Error = AttributeError>,
+{
+    type Error = AttributeError;
+    fn try_from(value: Kind) -> Result<Self, Self::Error> {
+        match value {
+            Kind::StructValueStatic(v) => v
+                .into_iter()
+                .map(|(k, v)| Ok((*k, v.try_into()?)))
+                .collect(),
+            _ => Err(AttributeError::ConversionImpossibleError),
+        }
+    }
+}
+impl<V> TryFrom<&Kind> for BTreeMap<&'static str, V>
+where
+    V: for<'a> std::convert::TryFrom<&'a Kind, Error = AttributeError>,
+{
+    type Error = AttributeError;
+    fn try_from(value: &Kind) -> Result<Self, Self::Error> {
+        println!("Obj {:?}", value);
+        match value {
+            Kind::StructValueStatic(v) => v
+                .into_iter()
+                .map(|(k, v)| Ok((*k, v.try_into()?)))
+                .collect(),
+            _ => Err(AttributeError::ConversionImpossibleError),
+        }
+    }
+}
+
+impl<T> TryFrom<&Kind> for Vec<T>
+where
+    T: for<'a> std::convert::TryFrom<&'a Kind, Error = AttributeError>,
+{
+    type Error = AttributeError;
+    fn try_from(value: &Kind) -> Result<Self, Self::Error> {
+        match value {
+            Kind::ListValueStatic(v) => v.iter().map(|v| v.try_into()).collect(),
+            _ => Err(AttributeError::ConversionImpossibleError),
+        }
+    }
+}
+
+impl<T> TryFrom<Kind> for Vec<T>
+where
+    T: for<'a> std::convert::TryFrom<&'a Kind, Error = AttributeError>,
+{
+    type Error = AttributeError;
+    fn try_from(value: Kind) -> Result<Self, Self::Error> {
+        match value {
+            Kind::ListValueStatic(v) => v.iter().map(|v| v.try_into()).collect(),
+            _ => Err(AttributeError::ConversionImpossibleError),
+        }
+    }
+}
+
+impl TryFrom<&Kind> for Kind {
+    type Error = AttributeError;
+    fn try_from(value: &Kind) -> Result<Self, Self::Error> {
+        match value {
+            Kind::BoolValue(v) => Ok(Kind::BoolValue(*v)),
+            Kind::NullValue(v) => Ok(Kind::NullValue(*v)),
+            Kind::StringValueStatic(v) => Ok(Kind::StringValueStatic(v)),
+            Kind::NumberValue(v) => Ok(Kind::NumberValue(*v)),
+            Kind::ListValueStatic(v) => Ok(Kind::ListValueStatic(v)),
+            _ => Err(AttributeError::ConversionImpossibleError),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub enum Kind {
+    NullValue(i32),
```

Comment:
Why does a `NullValue` contain an `i32`?

Reply:
this is a protobuf thing :)

Reply:
OK.